File: Merge Two Sorted Lists.cpp
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        if(list1 == NULL)
            return list2;
        if(list2 == NULL)
            return list1;
        ListNode *answer = NULL;
        if(list1->val <= list2->val){
            answer = list1;
            answer->next = mergeTwoLists(list1->next, list2);
        }
        else{
            answer = list2;
            answer->next = mergeTwoLists(list1, list2->next);
        }
        return answer;
    }
};

File: Merge Two Sorted Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        merged = ListNode()
        temp1 = merged

        while list1 and list2:
            if list1.val <= list2.val:
                temp2 = ListNode(list1.val)
                temp1.next = temp2
                temp1 = temp2
                list1 = list1.next
            else:
                temp2 = ListNode(list2.val)
                temp1.next = temp2
                temp1 = temp2
                list2 = list2.next
        
        while list1:
            temp2 = ListNode(list1.val)
            temp1.next = temp2
            temp1 = temp2
            list1 = list1.next

        while list2:
            temp2 = ListNode(list2.val)
            temp1.next = temp2
            temp1 = temp2
            list2 = list2.next
        
        return merged.next
